model-api-gen-gradle-test/kotlin-generation/src/test/kotlin/org/modelix/modelql/typed/TypedModelQLTest.kt

```diff
@@ -195,4 +236,167 @@ class TypedModelQLTest {
+            val descendants = root.children("classes").ofConcept(C_ClassConcept)
+                .member
+                .ofConcept(C_StaticMethodDeclaration)
+                .descendants()
+
+            val target = descendants.ofConcept(C_ParameterDeclaration)
+                .filter { it.name.equalTo(expected) }
+                .first()
+
+            descendants.ofConcept(C_VariableReference)
+                .first()
+                .setVariableDeclaration(target)
+        }
+
+        val actual = client.query { root ->
+            root.children("classes").ofConcept(C_ClassConcept)
+                .member
+                .ofConcept(C_StaticMethodDeclaration)
+                .descendants()
+                .ofConcept(C_VariableReference)
+                .first()
+        }
+        assertNotEquals(expected, oldValue)
+        assertEquals(expected, actual.variableDeclaration.name)
+    }
+
+    @Test
+    fun `set reference - null`() = runTest { httpClient ->
+        val client = ModelQLClient.builder().url("http://localhost/query").httpClient(httpClient).build()
+        val oldValue = client.query { root ->
+            root.children("imageGen").ofConcept(C_ImageGenerator).first().node
+        }
+
+        client.query { root ->
+            root.children("imageGen").ofConcept(C_ImageGenerator).first().setNode(null)
+        }
+
+        val actual = client.query { root ->
+            root.children("imageGen").ofConcept(C_ImageGenerator).first().node_orNull
+        }
+        assertNotNull(oldValue)
+        assertNull(actual)
+    }
+
+    @Test
+    fun `add new child`() = runTest { httpClient ->
+        val client = ModelQLClient.builder().url("http://localhost/query").httpClient(httpClient).build()
+
+        val oldNumChildren = client.query { root ->
+            root.children("classes").ofConcept(C_ClassConcept)
+                .first()
+                .member
+                .ofConcept(C_StaticMethodDeclaration)
+                .count()
+        }
+
+        client.query { root ->
+            root.children("classes").ofConcept(C_ClassConcept)
+                .first()
+                .addMember(C_StaticMethodDeclaration)
+        }
+
+        val children = client.query { root ->
+            root.children("classes").ofConcept(C_ClassConcept)
+                .first()
+                .member
+                .ofConcept(C_StaticMethodDeclaration)
+                .toList()
+        }
+
+        assertEquals(oldNumChildren + 1, children.size)
+        assertEquals(C_StaticMethodDeclaration.untyped().getUID(), children.last().untyped().concept?.getUID())
+    }
+
+    @Test
+    fun `add new child - default concept`() = runTest { httpClient ->
+        val client = ModelQLClient.builder().url("http://localhost/query").httpClient(httpClient).build()
+        client.query { root ->
+            root.descendants().ofConcept(C_XmlComment)
+                .first()
+                .addLines()
```

Comment:
The generated method name is a bit confusing in cases like this. Maybe `addTo` would be a better prefix?